torchvision/prototype/transforms/_augment.py

```diff
@@ -243,11 +239,12 @@ def _copy_paste(
         if blending:
             paste_alpha_mask = F.gaussian_blur(paste_alpha_mask.unsqueeze(0), kernel_size=[5, 5], sigma=[2.0])
 
+        inverse_paste_alpha_mask = ~paste_alpha_mask
         # Copy-paste images:
-        image = (image * (~paste_alpha_mask)) + (paste_image * paste_alpha_mask)
+        image = (image * inverse_paste_alpha_mask).add_(paste_image * paste_alpha_mask)
```

Comment:
Simply cache `~paste_alpha_mask` to avoid repeated estimations and use of in-place addition on the intermediate results. I don't have benchmarks for this because I thought it's quite uncontroversial.

Reply:
Nit: I would prefer not to mix the regular operators like `*` and `+` with the flow interface like `.add_` and `.mul_`. Since the regular operators don't support inplace operations on the right side of a statement, this means I would prefer using the flow style whenever inplace operations are present

```py
image = image.mul(inverse_paste_alpha_mask).add_(paste_image.mul(paste_alpha_mask))
```

Reply:
Non-blocking: We could perform inplace operations on `paste_image` in case we needed to resize it above. Otherwise it is one of the inputs and thus we need to do out of place operations like it is done here.